osu.Server.Spectator/Hubs/SpectatorHub.cs

```diff
@@ -30,13 +81,44 @@ public async Task BeginPlaySession(SpectatorState state)
 
         public async Task SendFrameData(FrameDataBundle data)
         {
-            await Clients.Group(GetGroupId(CurrentContextUserId)).UserSentFrames(CurrentContextUserId, data);
+            using (var usage = await GetOrCreateLocalUserState())
+            {
+                var score = usage.Item?.Score;
+                Debug.Assert(score != null);
+
+                score.ScoreInfo.Statistics = data.Header.Statistics;
+                score.ScoreInfo.MaxCombo = data.Header.MaxCombo;
+                score.ScoreInfo.Combo = data.Header.Combo;
```

Comment:
Just going to note that the legacy encoder will also read `TotalScore` and write it to the score, and I'm not seeing it be set anywhere in this diff, which means that in some cases it will be 0 rather than the actual value. It may be an issue if the replay is to ever be imported into the client in some way, possibly irrelevant for now.

Reply:
Apparently this isn't possible right now because we need to know the max combo of a beatmap to calculate score. Could potentially be retrieved from the database on the fly here, but not sure about it yet. I'll add a TODO.

This is less relevant for lazer because it will calculate the total score post replay loading, and I'm not sure yet if we even want these replays to support loading into stable (where it would be used).